peps/pep-9999.rst

```diff
@@ -0,0 +1,289 @@
+   each time they wish to update their project's development status.
+   Furthermore, because distributions are *de facto* immutable in the Python
+   packaging ecosystems, older distributions can't have their classifiers
+   updated to reflect the current status of the project.
+
+2. Indices can mark distributions and releases as "yanked", as originally
+   specified in :pep:`592`. Yanked distributions are not considered
+   eligible for dependency resolution.
+
+   When a distribution has been yanked, it is marked with ``data-yanked``
+   in the HTML index and with ``yanked: bool | str`` in the JSON index.
+   Additionally, indices like PyPI will hide yanked distributions by default
+   and will render them with a notice when the user navigates directly to them.
+
+   Yanking is machine-readable like trove classifiers, but is single-purpose
+   rather than general-purpose: users can specify a free-form "yank reason,"
+   but the semantics of yanking are fixed.
+
+3. PyPI itself has *project statuses*, which apply to the entire project
+   (i.e., all releases and distributions). Project statuses have both
+   maintainer- and index-admin-controllable states:
+
+   * PyPI administrators can "quarantine" a project. Quarantine behaves like
+     a strengthened yank: the project remains entirely uninstallable while
+     quarantined.
+
+   * Project maintainers can "archive" a project. Archiving a project
+     disables new release and distribution uploads to that project,
+     but otherwise has no effect on the project's downloadability.
+
+   Project statuses are machine-readable *in principle*, but are not currently
+   exposed via any of PyPI's APIs. Instead, PyPI renders project statuses on
+   each project's user-facing (i.e. non-index) HTML.
+
+In summary: there are multiple ways to communicate the "status" of a project in
+Python packaging, but none of these mechanisms are *simultaneously* (1)
+machine-readable, (2) general, (3) index-agnostic, and (4) updatable without
+pushing a new distribution:
+
+.. csv-table::
+    :header: "Mechanism", "Machine-readable?", "General?", "Index-agnostic?", "Updatable?"
+
+    "Trove classifiers", "✅", "✅", "✅", "❌"
+    "Yanking", "✅", "❌", "✅", "✅"
+    "Project statuses", "✅", "✅", "❌", "✅"
+
+This PEP proposes adopting PyPI's project statuses as an index-agnostic
+mechanism, satisfying all four conditions.
+
+Specification
+=============
+
+This PEP specifies two aspects: a set of project status markers,
+as well as their presentation in the standard HTML and JSON indices.
+
+Project status markers
+----------------------
+
+This PEP proposes the following project status markers.
+
+A project **MAY** have multiple status markers. When evaluating multiple status
+markers for index behavior, the index **MUST** use the strongest negative
+requirement. For example, if a project is both archived and quarantined, the
+index **MUST NOT** offer any distributions of the project for download.
```

Comment:
Flagging: I included this as a MAY, even though at the moment PyPI only supports a single status marker. I could go either way on it: having multiple markers (at least at the representation level) _could_ be useful, but it also significantly complicates the PEP (e.g. with this monotonic prescription).

Another reason to allow multiple statuses is because indices might want to include their own (e.g. `x-pypi-blahblah`), although I'm not aware of any actual demand for this.

Curious what others think.